livestream-compose-sample/src/main/java/io/getstream/livestream/compose/ui/LiveStreamHomeActivity.kt

```diff
@@ -0,0 +1,120 @@
+    private val factory by lazy {
+        ChannelViewModelFactory(
+            ChatClient.instance(), ChatDomain.instance(),
+            QuerySort.desc("last_updated"),
+            Filters.and(
+                Filters.eq("type", "livestream"),
+                Filters.`in`("members", listOf(ChatClient.instance().getCurrentUser()?.id ?: ""))
+            )
+        )
+    }
+
+    private val listViewModel: ChannelListViewModel by viewModels { factory }
+
+    @ExperimentalMaterialApi
+    @ExperimentalFoundationApi
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            var expanded by remember { mutableStateOf(false) }
+            var isDarkMode by remember { mutableStateOf(false) }
+            var isGrid by remember { mutableStateOf(true) }
+
+            ChatTheme(colors = if (isDarkMode) darkColorPalette() else lightColorPalette()) {
+                CustomHeader(
+                    title = resources.getString(R.string.app_name),
+                    actions = {
+                        //Adds a icon button for switching theme
+                        IconButton(
+                            onClick = {
+                                expanded = !expanded
+                            }) {
+                            Image(
+                                painter = painterResource(id = R.drawable.ic_theme_switch),
+                                contentDescription = stringResource(id = R.string.accessibilitySwitchTheme),
+                                modifier = Modifier
+                                    .width(24.dp)
+                                    .height(24.dp)
+                            )
+                        }
+                        //Theme chooser drop down selector
+                        DropdownMenu(
+                            expanded = expanded,
+                            onDismissRequest = { expanded = false }
+                        ) {
+
+                            DropdownMenuItem(onClick = {
+                                expanded = !expanded
+                                isDarkMode = false
+                            }) {
+                                Text(text = "LiveStream light theme")
+                            }
+                            DropdownMenuItem(onClick = {
+                                expanded = !expanded
+                                isDarkMode = true
+                            }) {
+                                Text(text = "LiveStream dark theme")
+                            }
+                        }
+
+                        //Adds a icon button for switching grid toggle
+                        IconButton(onClick = {
+                            isGrid = !isGrid
+                        }) {
+                            Icon(
+                                painter = painterResource(id = R.drawable.ic_toggle_grid),
```

Comment:
we should toggle the icon too, based on the `isGrid` state.

Reply:
Added functionality to capture this behaviour